external/kafka/src/main/scala/org/apache/spark/streaming/kafka/KafkaUtils.scala

```diff
@@ -399,7 +418,7 @@ object KafkaUtils {
     val kc = new KafkaCluster(kafkaParams)
     val reset = kafkaParams.get("auto.offset.reset").map(_.toLowerCase)
 
-    (for {
+    KafkaCluster.checkErrors(for {
```

Comment:
that checkErrors method is just giving a name to the repeated pattern of folding over the EIther returned from the kafka cluster api, to either unwrap it to a result or throw an exception.